src/ucp/core/ucp_am.c

```diff
@@ -921,7 +991,7 @@ static ucs_status_t ucp_am_long_first_handler(void *am_arg, void *am_data,
         ucs_queue_del_iter(&ep_ext->am.mid_rdesc_q, iter);
         ucp_am_copy_data_fragment(first_rdesc, mid_hdr + 1,
                                   mid_rdesc->length - sizeof(*mid_hdr),
-                                  mid_hdr->offset + sizeof(*first_hdr));
+                                  mid_hdr->offset + first_rdesc->payload_offset);
```

Comment:
unrelated fix?

Reply:
yes, but zcopy tests fail without it (when first fragment arrives later than second)